provider/confluence/index.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+    ItemsParams,
+    ItemsResult,
+    MentionsParams,
+    MentionsResult,
+    MetaParams,
+    MetaResult,
+    Provider,
+} from '@openctx/provider'
+import { type Page, listPages } from './api.js'
+
+export type Settings = {
+    host: string
+    port?: string
+    email: string
+    apiToken: string
+}
+
+const confluenceProvider: Provider = {
+    meta(params: MetaParams, settings: Settings): MetaResult {
+        return { name: 'Confluence Pages', mentions: {} }
+    },
+
+    async mentions(params: MentionsParams, settings: Settings): Promise<MentionsResult> {
+        const spaces = await listPages(settings, params.query)
+        return spaces.map(page => {
+            return {
+                title: page.title,
+                uri: page.uri,
+                description: page.space.name,
+                ui: {
+                    hover: {
+                        text: page.title,
+                    },
```

Comment:
is it worth including the hover text here given it is just the same value as title?